dynamixel_driver/src/dynamixel_driver/dynamixel_io.py

```diff
@@ -83,7 +83,8 @@ def close(self):
             self.ser.flushOutput()
             self.ser.close()
 
-    def __write_serial(self, data):
+    def __write_serial(self, packet):
+        data = array('B', packet)
```

Comment:
I refactored all uses of `__write_serial` to pass in the list of integers directly, instead of having the functions try to serialize the data before passing it in. This made it easier to isolate serialization issues.